BlockEntity/BEBerryBush.cs

```diff
@@ -188,6 +188,7 @@ private void CheckGrow(float dt)
         public override void OnExchanged(Block block)
         {
             base.OnExchanged(block);
+            transitionHoursLeft = GetHoursForNextStage();

```

Comment:
Moving it here makes sure that it always runs when the block gets exchanged, which helps with making sure it runs after harvest now, but also fixes it so raccoons harvesting will properly reset the timer like you would expect.